fission/src/systems/simulation/wpilib_brain/WPILibBrain.ts

```diff
@@ -231,3 +281,53 @@ export class SimMapUpdateEvent extends Event {
 }
 
 export default WPILibBrain
+
+abstract class SimOutputGroup {
+    public name: string
+    public ports: number[]
+    public drivers: Driver[]
+    public type: SimType
+
+    public constructor(name: string, ports: number[], drivers: Driver[], type: SimType) {
+        this.name = name
+        this.ports = ports
+        this.drivers = drivers
+        this.type = type
+    }
+
+    public abstract Update(deltaT: number): void
+}
+
+export class PWMGroup extends SimOutputGroup {
+    public constructor(name: string, ports: number[], drivers: Driver[]) {
+        super(name, ports, drivers, "PWM")
+    }
+
+    public Update(_deltaT: number) {
+        // let average = 0;
+        for (const port of this.ports) {
+            const speed = SimPWM.GetSpeed(`${port}`) ?? 0
+            // average += speed;
+            console.log(port, speed)
+        }
+        // average /= this.ports.length
+
+        // this.drivers.forEach(d => {
+        //     (d as WheelDriver).targetWheelSpeed = average * 40
+        //     d.Update(_deltaT)
+        // })
+    }
+}
```

Comment:
Yeah, this will be done within tasks for specific types of motor support. The scope of this PR is just parse the websocket data and make it easily available for motors/motor types that will be implemented later on.